python/usepyhf/confidence.py

```diff
@@ -0,0 +1,39 @@
+import pyhf
+import numpy as np
+from usepyhf import util
+
+def upper_limit(workspace:pyhf.Workspace):
+    """
+    Calculates the observed upper limit of the POI at the 95% CL.
+    This function also prints the expected limit median and +/- [1, 2] sigma.
+    """
+    #Extract model and data
+    model = workspace.model()
+    data = workspace.data(model)
+    
+    #Find limits
+    poi_index = model.config.poi_index
+    poi_min, poi_max = model.config.suggested_bounds()[poi_index]
+    poi_name = model.config.par_order[poi_index]
+
+    #Print info
+    print("<INFO> pyhf: Finding upper limits:")
+    print(f"Parameter of interest: {poi_name}")
+    print(f"Suggested bounds: { poi_min, poi_max }")
+
+    #Fix the parameter "SigXsec" if it exists
+    fix_pars = util.fix_sigxsec(model)
+
+    #Calculate upper limits
+    poi_values = np.linspace(poi_min, poi_max, 50)
```

Comment:
Instead of 50, the number of points can be taken from nPoints in the config manager: https://github.com/histfitter/histfitter/blob/master/python/configManager.py#L112